### ISSUE TITLE
Hold for me and call screen not working

### ISSUE THREAD

**Issue by drandrew1983** [jacopotediosi/GoogleDialerMod#7]
Dear jacopotediosi,
Thank you so much for your work and support.
actually I have a problem activating hold for me and call screen. I followed your instructions but it always shows activating (require WiFi) and never get activated.
please check the screenshot
![Screenshot_20210930-134324](https://user-images.githubusercontent.com/33606175/137413985-514b7fac-9d04-416e-9f73-97038b9b73cf.jpg)
![Screenshot_20210930-134319](https://user-images.githubusercontent.com/33606175/137413988-d7e88f71-66f7-4312-a1b7-a0840e4b7ceb.jpg)

**Comment by hehedemuse** [comment]
click "see how call screen works"

**Comment by drandrew1983** [comment]
> 

actually call screen is behaving erratically, sometimes it works sometimes I have to revert back the settings and activate it again in Google dialer mod.
but hold for me never worked, was wondering if it works only on google Pixel phones

**Comment by hehedemuse** [comment]
Yes

**Comment by jacopotediosi** [comment]
Hi,
I know it's been a long time. I recently started updating GoogleDialerMod again.

Regarding the Call Screen feature, I just released the [version 1.05](https://github.com/jacopotediosi/GoogleDialerMod/releases/tag/v1.05) which should allow you to activate it correctly.

About Hold for Me, instead, I still need some more time to debug why it doesn't work.

**Comment by JJsheep11** [comment]
> Hi, I know it's been a long time. I recently started updating GoogleDialerMod again.
> 
> Regarding the Call Screen feature, I just released the [version 1.05](https://github.com/jacopotediosi/GoogleDialerMod/releases/tag/v1.05) which should allow you to activate it correctly.
> 
> About Hold for Me, instead, I still need some more time to debug why it doesn't work.

I have tried the new version (1.05), but call screen always shows activating (require WiFi) and never get activated.

**Comment by jacopotediosi** [comment]
> I have tried the new version (1.05), but call screen always shows activating (require WiFi) and never get activated.

Uhm it may be a stupid question, but have you tried to force close the Dialer, activate wifi and reopen it? It really needs wifi.

Also, what country are you in and what language is set on your phone?

**Comment by JJsheep11** [comment]
> > I have tried the new version (1.05), but call screen always shows activating (require WiFi) and never get activated.
> 
> Uhm it may be a stupid question, but have you tried to force close the Dialer, activate wifi and reopen it? It really needs wifi.
> 
> Also, what country are you in and what language is set on your phone?

Yes, I did all of them. I am outside of US, and using English (US). My phone is Pixel 7.

**Comment by shadyrexman** [comment]
Same here, the older version of google phone worked perfectly but after the latest update, the call screen stopped working and I tried to force close and restart my phone many times ,and it didn't work. After I used the newest mod and it keeps saying activating. But using the old build of google phone works though.

**Comment by jacopotediosi** [comment]
Could someone please share his logcat captured while viewing the "activating" screen, after having started the Dialer in debug mode with `adb shell -c 'am start -D com.google.android.dialer'` command?
Also, I need to know:
- In which country you are
- Which language have you set in your phone
- Your Google Dialer version

**Comment by shadyrexman** [comment]
> > Same here, the older version of google phone worked perfectly but after the latest update, the call screen stopped working and I tried to force close and restart my phone many times ,and it didn't work. After I used the newest mod and it keeps saying activating. But using the old build of google phone works though.
> 
> Which Google Dialer version are you using?

Right now I'm using the old build 94.0.487380581- publicbeta and I can use call screening , but the layout of the settings is different

**Comment by jacopotediosi** [comment]
> but the layout of the settings is different in new build

Yes, it is not because of a Dialer update, but because there are two call screening features:

- A basic one, managed by `call_screen` and `speak_easy` flags, which uses a separate "Call Screen" menu in Dialer settings
- An enhanced version, managed by the `revelio` flag (which requires also the basic call screen to be enabled), which merges with the already existing "ID/Spam" entry in Dialer settings

The suggested mod switch in GoogleDialerMod v1.05 enables the enhanced one.

**Comment by jacopotediosi** [comment]
Currently it seems that Call Screen may not work in some countries (SIM-based): https://github.com/jacopotediosi/GoogleDialerMod/issues/42#issuecomment-1326709693
I'll try to find a workaround

**Comment by shadyrexman** [comment]
> Currently it seems that Call Screen may not work in some countries (SIM-based): https://github.com/jacopotediosi/GoogleDialerMod/issues/42#issuecomment-1326709693
> I'll try to find a workaround

Thank you so much for this.

**Comment by jacopotediosi** [comment]
Hi!
Call Screen should work with latest v1.06 (for more details please see #42)
I'll work on Hold for Me soon.

**Comment by drandrew1983** [comment]
Dear Jacob,It works fine for me now.waiting for "Hold for me", hope you can activate it.Thank you so much for your work 
    On Sunday, November 27, 2022 at 03:24:10 PM GMT+5, Jacopo Tediosi ***@***.***> wrote:  
 
 


Hi!
Call Screen should work with latest v1.06 (for more details please see #42)
I'll work on Hold for Me soon.

**Comment by Yoanndp** [comment]
Hello Jacob, I did some logging and there's what I have found on my phone (Pixel 7, HFM enabled from GDMod) :
```
11-29 08:27:53.496  6347  6402 I DialerTelephonyManagerCache: com.google.android.dialershared.telephony.TelephonyManagerCache.lambda$setUpNetworkMemoize$0:83 getNetworkOperatorName
11-29 08:27:53.498  6347  6402 I DialerTelephonyManagerCache: com.google.android.dialershared.telephony.TelephonyManagerCache.lambda$setUpNetworkMemoize$1:92 getNetworkCountryIso
11-29 08:27:53.498  6347  6402 I DialerTelephonyManagerCache: com.google.android.dialershared.telephony.TelephonyManagerCache.lambda$setUpNetworkMemoize$2:101 getNetworkOperator
11-29 08:27:53.500  6347  6402 I DialerTelephonyManagerCache: com.google.android.dialershared.telephony.TelephonyManagerCache.lambda$setUpNetworkMemoize$4:121 getSimCountryIso
11-29 08:27:53.502  6347  6405 D vulkan  : searching for layers in '/data/app/~~3k-Bq0A3wPtIf3DVSD1APA==/com.google.android.dialer-eODAIdVVeazpRsYKoeHf4w==/lib/arm64'
11-29 08:27:53.508  6347  6405 D vulkan  : searching for layers in '/data/app/~~3k-Bq0A3wPtIf3DVSD1APA==/com.google.android.dialer-eODAIdVVeazpRsYKoeHf4w==/base.apk!/lib/arm64-v8a'
11-29 08:27:53.558  6347  6431 I DialerAudioFileSuperpackProvider: com.google.android.libraries.micore.telephony.common.audiomessages.AudioFileSuperpackProvider.getFileInternal:316 Providing file /data/user/0/com.google.android.dialer/files/superpacks/call_screen_audio_messages/1613501903_en-us_cloud-en-US-Wavenet-F_71c80e40489676cc808b1e9df1b5b916/71c80e40489676cc808b1e9df1b5b916.opus from superpack
11-29 08:27:53.559  6347  6431 I DialerAudioFileSuperpackProvider: com.google.android.libraries.micore.telephony.common.audiomessages.AudioFileSuperpackProvider.getFileInternal:316 Providing file /data/user/0/com.google.android.dialer/files/superpacks/call_screen_audio_messages/1613501903_en-us_cloud-en-US-Wavenet-F_dc43808541a81bb4e880a0f246d8ce3b/dc43808541a81bb4e880a0f246d8ce3b.opus from superpack
11-29 08:27:53.635  6347  6440 I DialerAudioFileSuperpackProvider: com.google.android.libraries.micore.telephony.common.audiomessages.AudioFileSuperpackProvider.getFileInternal:316 Providing file /data/user/0/com.google.android.dialer/files/superpacks/call_screen_audio_messages/1613501903_en-us_cloud-en-US-Wavenet-F_e2f607685cf332b687aa5ca8fef61545/e2f607685cf332b687aa5ca8fef61545.opus from superpack
11-29 08:27:53.635  6347  6440 I DialerAudioFileSuperpackProvider: com.google.android.libraries.micore.telephony.common.audiomessages.AudioFileSuperpackProvider.getFileInternal:316 Providing file /data/user/0/com.google.android.dialer/files/superpacks/call_screen_audio_messages/1613501903_en-us_cloud-en-US-Wavenet-F_dede97e191c3d1b9389f44851ac7838d/dede97e191c3d1b9389f44851ac7838d.opus from superpack
11-29 08:27:53.636  6347  6440 I DialerAudioFileSuperpackProvider: com.google.android.libraries.micore.telephony.common.audiomessages.AudioFileSuperpackProvider.getFileInternal:316 Providing file /data/user/0/com.google.android.dialer/files/superpacks/call_screen_audio_messages/1613501903_en-us_cloud-en-US-Wavenet-F_8a1bfe69068649e82f270263888a69f3/8a1bfe69068649e82f270263888a69f3.opus from superpack
11-29 08:27:53.637  6347  6440 I DialerAudioFileSuperpackProvider: com.google.android.libraries.micore.telephony.common.audiomessages.AudioFileSuperpackProvider.getFileInternal:316 Providing file /data/user/0/com.google.android.dialer/files/superpacks/call_screen_audio_messages/1613501903_en-us_cloud-en-US-Wavenet-F_b53062dd3b2119adb3992d7e9fba9174/b53062dd3b2119adb3992d7e9fba9174.opus from superpack
```
It's still trying to use en_US files (`en-us_cloud-en-US`) even if my SIM locale is fr_BE. I can provide you full log if needed.

**Comment by drandrew1983** [comment]
Dear Jacob, can you please create a setting to activate this feature as well? It's called Direct my call.
I don't know how I enabled it, I'm just playing with the toggles
![Screenshot_2022-12-19-16-07-28-74_54063fc6848ac98a54ce5f2a63661433](https://user-images.githubusercontent.com/33606175/208413340-d73f5db7-edfc-4db4-9b9a-d1414be9e3d1.jpg)

**Comment by jacopotediosi** [comment]
> Dear Jacob, can you please create a setting to activate this feature as well? It's called Direct my call. I don't know how I enabled it, I'm just playing with the toggles ![Screenshot_2022-12-19-16-07-28-74_54063fc6848ac98a54ce5f2a63661433](https://user-images.githubusercontent.com/33606175/208413340-d73f5db7-edfc-4db4-9b9a-d1414be9e3d1.jpg)

Yeah, the internal name for this feature is "xatu". I'll work on it, but please expect long delays because I'm currently working on other parts of GoogleDialerMod.

**Comment by fillwithjoy1** [comment]
> > Dear Jacob, can you please create a setting to activate this feature as well? It's called Direct my call. I don't know how I enabled it, I'm just playing with the toggles ![Screenshot_2022-12-19-16-07-28-74_54063fc6848ac98a54ce5f2a63661433](https://user-images.githubusercontent.com/33606175/208413340-d73f5db7-edfc-4db4-9b9a-d1414be9e3d1.jpg)
> 
> Yeah, the internal name for that feature is "xatu". I'll work on it, but please expect long delays because I'm currently working on other parts of GoogleDialerMod.


Which numbers can I call that support direct my call?

**Comment by jacopotediosi** [comment]
> Which numbers can I call that support direct my call?

I don't know yet

**Comment by fillwithjoy1** [comment]
> > Which numbers can I call that support direct my call?
> 
> I don't know yet

It's funny because there are US numbers like verizon that work with Direct my Call, but it isn't working here in Australia. Maybe it's locked down to United States?

**Comment by gerhard101-byte** [comment]
Having the same issue with hold for me. Not working it's just saying wait to activate with Wi-Fi.
Also direct my call is on but have had no success yet
Call screen works great however call recordings also only work when silence call recordings is off.

My language is US and I am based in South Africa

**Comment by fillwithjoy1** [comment]
Version 100.0.512999549
GoogleDialerMod v2

Phone is stuck on activating direct my call

**Comment by jacopotediosi** [comment]
Again, Direct My Call and Hold for Me are currently not supported yet. Please stop reporting they aren't working.
